File: app/src/main/java/com/doubean/ford/data/vo/RecommendedGroupsResult.java
package com.doubean.ford.data.vo;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.List;

@Entity(tableName = "recommended_groups_result")
public class RecommendedGroupsResult {
    @NonNull
    public final List<Long> ids;

    @PrimaryKey
    @NonNull
    public GroupRecommendationType type;

    public RecommendedGroupsResult(@NonNull GroupRecommendationType type, @NonNull List<Long> ids) {
        this.type = type;
        this.ids = ids;
    }
}

File: app/src/main/java/com/doubean/ford/data/db/model/RecommendedGroupsResult.kt
package com.doubean.ford.data.db.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.doubean.ford.model.GroupRecommendationType

@Entity(tableName = "recommended_groups_results")
class RecommendedGroupsResult(
    @PrimaryKey
    @ColumnInfo("recommendation_type")
    val recommendationType: GroupRecommendationType,
    val ids: List<String>,
)